### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by MWFlood257**
Bugs flagged in testing all fixed

### CODE REVIEW

File: src/datasource.ts [hunk 1]
```diff
@@ -132,31 +121,56 @@ export class KDBDatasource {
...
         console.log('TEMPLATESRV:', this.templateSrv);
         console.log('VARIABLES: ', instVariables);
+        console.log('scp',scoped)
         let scopedVarArray = Object.keys(scoped);
         let scopedValueArray = [];
         //scoped variables inject
         for(let k = 0; k < scopedVarArray.length; k++) {
             scopedValueArray.push(scoped[scopedVarArray[k]].value);
-            scopedVarArray[k] = "$" + scopedVarArray[k];
+            scopedVarArray[k] = "$" + '{' + scopedVarArray[k] + '}';
```
Comment (DGaquaq) [inline src/datasource.ts:150]:
```suggestion
            scopedVarArray[k] = "${" + scopedVarArray[k] + '}';

```

File: src/datasource.ts [hunk 2]
```diff
@@ -132,31 +121,56 @@ export class KDBDatasource {
...
         };
         //local variables inject (user variables)
         for(let i = 0; i < instVariables.length; i++) {
-            let varname = '$' + instVariables[i].name
+            let varname = '$' + '{' + instVariables[i].name + '}'
```
Comment (DGaquaq) [inline src/datasource.ts:154]:
```suggestion
            let varname = '${' + instVariables[i].name + '}'

```

File: Demo/VariableDemo_v2/DemoDatabase/DummyDataSource.bat [hunk 3]
```diff
@@ -0,0 +1,3 @@
+@echo off
```
Comment (CallumOrmerod) [inline Demo/VariableDemo_v2/DemoDatabase/DummyDataSource.bat:1]: this file shouldn't be in the PR

File: Demo/VariableDemo_v2/DemoDatabase/fh.q [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* q fh.q */
```
Comment (CallumOrmerod) [inline Demo/VariableDemo_v2/DemoDatabase/fh.q:1]: this file shouldn't be in the PR

File: Demo/VariableDemo_v2/DemoDatabase/pubsub.q [hunk 5]
```diff
@@ -0,0 +1,46 @@
+\p 5001
```
Comment (CallumOrmerod) [inline Demo/VariableDemo_v2/DemoDatabase/pubsub.q:1]: this file shouldn't be in the PR

File: src/response_parser.ts [hunk 6]
```diff
@@ -101,14 +101,18 @@ export default class ResponseParser {
...
         
         //looop for each grouping(sym)*************
         for (let g = 0; g < response.payload[0].length; g++) {
             var curCol = 2;
             //looping through columns if multiple have been selected
             for (curCol = 2; curCol <= colKeys.length; curCol++) {
-                var fieldName = (req[1].queryParam.query.type == "`select") ? req[1].queryParam.column[curCol - 2][1] : colKeys[curCol - 1];
-
+                var fieldName = (req[1].queryParam.query.type == "`select") ? req[1].queryParam.column[curCol - 2][1].replace('`','') : colKeys[curCol - 1];
+                console.log('r', req[1].queryParam.column.length);
```
Comment (CallumOrmerod) [inline src/response_parser.ts:115]:
```suggestion
               // console.log('r', req[1].queryParam.column.length);

```

**Pull request merged**